### PR TITLE
Mark processes dead more robustly on startup

### PR DESCRIPTION
**Pull request by sam-github**
connected to strongloop-internal/scrum-nodeops#571

### CODE REVIEW

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
failed because log output didn't have "listening on"... but its hard to tell, in the DEBUG build, there are 100s of lines of output until the app output is seen. Could just be a race or a buffering issue.

I don't think its related, hopefully, the next test will pas.

@rmg, are you aware of any instability in that test?

```
# pmctl log using REST API
ok 280 cmd "status 1" pattern /Processes/
ok 281 exit status should be zero for: cmd "log-dump 1" pattern /.+ worker:1 pid \d+ listening on \d+/
not ok 282 cmd "log-dump 1" pattern /.+ worker:1 pid \d+ listening on \d+/
```

http://ci.strongloop.com/job/strong-pm.pr/601/label=amazon-64-0.12/consoleText

**Comment by rmg** [comment]
That test could fail if the test system was under heavy load and the worker hadn't actually started listening yet. Looking at the test app it looks like there are some semi-static strings printed immediately on startup that might make for better test strings.

**Comment by rmg** [comment]
LGTM.

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
Rerunning CI now that #206 has been merged. Should pass now.

**Comment by sam-github** [comment]
@slnode test please

File: lib/service-manager.js [hunk 1]
```diff
@@ -92,14 +92,20 @@ prototype.initOrUpdateDb = function(meshApp, callback) {
...
     });
   }
 
   function stopProcess(proc, callback) {
-    debug('mark stopped: pid %s wid %d', proc.pid, proc.workerId);
+    // Anything other than null, undefined, or '' is a reason.
+    if (proc.stopReason == null || proc.stopReason === '') {
+      console.log('NOT mark stopped: pid %j wid %j reason %j',
+            proc.pid, proc.workerId, proc.stopReason);
+      return callback();
```
Comment (rmg) [inline lib/service-manager.js]: I think the conditional is inverted.. `null` or `''` are the cases that _should_ be getting marked as stopped, aren't they?

Comment (sam-github) [inline lib/service-manager.js]: hm, I think you are right, I broke that in some last minute refactoring

**Pull request merged**